vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -238,7 +239,12 @@ def refresh_infrastructure_cache(self):
                 # Hosts are not considered as parents of the VMs they run, we use the `runtime.host` property
                 # to get the name of the ESXi host
                 runtime_host = properties.get("runtime.host")
-                runtime_host_props = infrastructure_data[runtime_host] if runtime_host else {}
+                runtime_host_props = {}  # type: InfrastructureDataItem
+                if runtime_host:
+                    if runtime_host in infrastructure_data:
+                        runtime_host_props = infrastructure_data.get(runtime_host, {}) if runtime_host else {}
```

Comment:
```suggestion
                        runtime_host_props = infrastructure_data.get(runtime_host, {})
```
Do you need to check `if runtime_host` twice?

Reply:
I want to make sure we don't end up in a situation where `runtime_host` is None and `infrastructure_data[None]` is set.
And we shouldn't log the debug log line with a None runtime_host

Reply:
But we already check it here https://github.com/DataDog/integrations-core/pull/10012/files#diff-8d730e4ffb87a6578a9413e77ef494317dcd057e8bcd88d3ae5a50a7629588c6R243 so when would it ever get to the `else {}`?

Reply:
I didn't understand your comment initially, yeah that's completely correct